submissions/crealime/html-css-popup/css/main.css

```diff
@@ -0,0 +1,267 @@
+  font-style: normal
+}
+
+html {
+  font-size: 16px;
+  overflow-x: hidden
+}
+
+body {
+  background-color: #dadada;
+  font-family: "Montserrat", sans-serif
+}
+
+* {
+  box-sizing: border-box
+}
+
+a {
+  color: #343434;
+  text-decoration: none
+}
+
+a:hover, a:focus {
+  color: #e74100
+}
+
+a:focus {
+  outline: none
+}
+
+.wrapper {
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+  min-height: 100vh
+}
+
+.header {
+  position: relative;
+  background: #fff;
+  z-index: 1
+}
+
+.top-menu {
+  height: 4.375rem;
+  padding: 0 1.25rem
+}
+
+.top-menu__nav {
+  height: inherit
+}
+
+.top-menu__ul {
+  height: inherit;
+  display: flex;
+  justify-content: flex-end;
+  align-items: center
+}
+
+.top-menu__li {
+  position: relative;
+  list-style-type: none;
+  margin-right: 0.625rem
+}
+
+.top-menu__li:last-child {
+  margin-right: 0
+}
+
+.top-menu__link {
+  display: block;
+  padding: 0.4375rem 0.625rem;
+  border: solid 1px transparent;
+  border-radius: 0.25rem;
+  transition-duration: 300ms
+}
+
+.top-menu__link:focus {
+  border: solid 1px #b4b4b4
+}
+
+.top-menu__icon {
+  width: 1.5rem;
+  height: 1.5rem;
+  object-fit: cover
+}
+
+.top-menu__avatar {
+  width: 2.25rem;
+  height: 2.25rem;
+  object-fit: cover;
+  border-radius: 50%
+}
+
+.top-menu__label {
+  display: block;
+  cursor: pointer;
+  border: solid 1px transparent;
+  border-radius: 0.25rem;
+  padding: 0.4375rem 0.625rem
+}
+
+.top-menu__input {
+  position: absolute;
+  appearance: none
+}
+
+.top-menu__input:focus ~ .top-menu__label {
+  border: solid 1px #b4b4b4
+}
+
+.top-menu__input:checked ~ .inner-menu {
+  visibility: visible;
+  opacity: 1;
+  pointer-events: auto
+}
+
+.top-menu__icon {
+  width: 1.5rem;
+  height: 1.5rem;
+  object-fit: cover
+}
+
+.inner-menu {
+  position: absolute;
+  top: 4.375rem;
+  right: 0;
+  visibility: hidden;
+  pointer-events: none;
+  opacity: 0;
+  transition-duration: 300ms;
+  z-index: 2
+}
+
+.inner-menu__ul {
+  background: #fff;
+  padding: 1.25rem;
+  display: grid;
+  min-width: 25rem;
+  grid-template-columns:repeat(3, 1fr)
+}
+
+.inner-menu__ul:before {
+  content: "";
+  position: absolute;
+  top: -1.75rem;
+  right: 0.4375rem;
+  border: 15px solid transparent;
+  border-bottom: 15px solid #fff;
+  z-index: -1
+}
+
+.inner-menu__li {
+  list-style-type: none
+}
+
+.inner-menu__li:nth-child(n+10) {
+  display: none
+}
+
+.inner-menu__link {
+  height: 100%;
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+  align-items: center;
+  padding: 0.625rem;
+  border: solid 1px transparent;
+  transition-duration: 300ms
+}
+
+.inner-menu__link:focus {
+  border: solid 1px #cdcdcd
+}
+
+.inner-menu__icon {
+  max-width: 100%;
+  max-height: 4.375rem;
+  margin-bottom: 0.625rem
+}
+
+.inner-menu__input {
+  position: absolute;
+  appearance: none
+}
+
+.inner-menu__input:checked ~ .inner-menu__ul .inner-menu__li:nth-child(n+10) {
+  display: block
```

Comment:
`:nth-child(n+10) ` is not an obvious and good decision, try to find a better way ;) If you will add one more icon seems it will crash our app

Reply:
Menu worked with any number of items, I checked. With :nth-child we could conveniently adjust the number of visible items. Now I've done it differently.
I placed the "More" button inside the menu, now focus goes to it after the ninth item. But it broke the menu structure (. It's a pity that in css there is no way to select the parent element ).
I used sass variables like:
`$color-prime: #343434`
`$brd-visible: solid 1px lighten($color-prime, 50%)`
`border: $brd-visible`
Added for example variables to the final css file in frontend-2022-homeworks repo.